### PR TITLE
Created Signup Page component

### PR DESCRIPTION
**Pull request by anilkumar057**
## Summary
- Create Signup Page component.

## Details
- Created `SignUp` component for 'Sign Up' page.
  - Defined `formData` useState variable to store the input fields value.
  - Defined `shouldShowError` useState variable to determine 'Create account' button is clicked, should show the validation errors.
  - Defined method `onFormInputChange`, call on input value change to update the 'formData'.
  - Defined method `handleSignUp` to handle the signup.
  - Created UI as per the figma design.

## Testing
Tested locally on development server

## Screenshots

Signup-page-without-error

Signup-page-with-errors

### CODE REVIEW

File: src/components/SignUp.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+const validationSchema = Yup.object().shape({
+  firstName: Yup.string().label('First Name').trim().max(25).required(),
+  lastName: Yup.string().label('Last Name').trim().max(25).required(),
+  email: Yup.string().label('Email').trim().email().required(),
+  password: Yup.string().label('Password').trim().min(4).max(25).required(),
+  confirmPassword: Yup.string()
+    .label('Confirm Password')
+    .trim()
+    .oneOf([Yup.ref('password')], 'Password must match'),
+});
```
Comment (nishantpal1911) [inline src/components/SignUp.tsx:19]: No need to trim the passwords. Leading / trailing spaces are mostly allowed in passwords.

File: src/components/SignUp.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  });
+  const [shouldShowError, setShouldShowError] = useState<boolean>(false);
+
+  const toast = useToast();
+  const { errors, validateField, validateForm } = useValidation<SchemaType>(validationSchema);
+
+  const onFormInputChange = (key: keyof SchemaType, value: SchemaType[keyof SchemaType]) => {
+    setFormData({ ...formData, [key]: value });
+    validateField(key, value);
+    console.log('errors', errors);
```
Comment (nishantpal1911) [inline src/components/SignUp.tsx:39]: Can remove `console.log` once testing is done and ready to commit.

**Review by nishantpal1911 — COMMENTED** [review]
Well written code! Just a couple nit-picks.

Also, to confirm, does the width of the form match the Figma [design](https://www.figma.com/design/9CoHc6iF6fi9bzXRjWdORz/Design?node-id=269-18385&t=ORNJr52nv8OZV95C-4)?

File: src/components/SignUp.tsx (hunk 1 above)
Comment (anilkumar057) [reply src/components/SignUp.tsx]: Removed trim method from validation schema for password and confirm password.

Thank you!

File: src/components/SignUp.tsx (hunk 2 above)
Comment (anilkumar057) [reply src/components/SignUp.tsx]: Removed console log. Thank you!

**Pull request merged**